pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarPartitionLevelConsumer.java

```diff
@@ -46,11 +47,18 @@ public class PulsarPartitionLevelConsumer extends PulsarPartitionLevelConnection
     implements PartitionGroupConsumer {
   private static final Logger LOGGER = LoggerFactory.getLogger(PulsarPartitionLevelConsumer.class);
   private final ExecutorService _executorService;
+  private final Reader _reader;
   private boolean _enableKeyValueStitch = false;
 
   public PulsarPartitionLevelConsumer(String clientId, StreamConfig streamConfig,
       PartitionGroupConsumptionStatus partitionGroupConsumptionStatus) {
-    super(clientId, streamConfig, partitionGroupConsumptionStatus.getPartitionGroupId());
+    super(clientId, streamConfig);
+    PulsarConfig config = new PulsarConfig(streamConfig, clientId);
+    _reader = createReaderForPartition(config.getPulsarTopicName(),
```

Comment:
We should also close `_reader` in `PulsarPartitionLevelConsumer.close` method

Reply:
good catch!